fix: don't reuse session across requests

m1racoli opened the pull request:
Concurrent requests will cause a conflict as they both use the same session.

Discovered this issue while pausing DAGs asynchronously.

m1racoli committed:
fix: don't reuse session across requests

Concurrent requests will cause a conflict as they both use the same session.

fritz-astronomer reviewed (approved):
lgtm, other than a refactor for the middleware / dependency injection thought, if it's feasible

m1racoli committed:
refactor: store per request session in application context

Using [Flask's Application Globals](https://flask.palletsprojects.com/en/stable/api/#application-globals), we can store and retrieve a per request session object.

m1racoli commented:
Ok, i know why tests fail. That flask.g object can only be accessed during a Flask request. Will need to mock `self.session` for tests.

m1racoli committed:
test: ensure tests are run within flask app context
